test/StoragelessSessionTest/Http/SessionMiddlewareTest.php

```diff
@@ -197,9 +194,34 @@ public function testSkipsInjectingSessionCookieOnEmptyContainer(SessionMiddlewar
         self::assertNull(FigResponseCookies::get($response, SessionMiddleware::DEFAULT_COOKIE)->getValue());
     }
 
-    public function testRemovesSessionCookieOnEmptySessionContainer()
+    /**
+     * @dataProvider validMiddlewaresProvider
+     */
+    public function testRemovesSessionCookieOnEmptySessionContainer(SessionMiddleware $middleware)
     {
-        self::markTestIncomplete('This feature is yet to be implemented, and we may do so in a different middleware');
+        $validTokenRequest = (new ServerRequest())
```

Comment:
We need an additional test where no cookie is set upfront, and no cookie removal is forced